### PR TITLE
chore: move pytest config to pyproject.toml

### PR DESCRIPTION
**Pull request by henryiii**
### Description

This moves (as mentioned elsewhere) and hardens the pytest config. I expect to
probably have to filter or fix warnings, so I don't really expect this to
immediately pass.

### Checklists

- [ ] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

**Comment by henryiii** [comment]
What should I do with:

```
pytest.PytestRemovedIn8Warning: Raising unittest.SkipTest to skip tests during collection is deprecated. Use pytest.skip() instead.
```

?

File: src/py/pyodide/http.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ async def unpack_archive(self, *, extract_dir=None, format=None):
 
 
 async def pyfetch(url: str, **kwargs) -> FetchResponse:
-    """Fetch the url and return the response.
+    r"""Fetch the url and return the response.
```
Comment (henryiii) [inline src/py/pyodide/http.py:211]: FYI, I couldn't run this due to `\*` later, which is not a valid Python escape sequence. I just added an r here, but might be worth checking to see what the normal way to write `**kwargs` in this style of docstring is.

Comment (henryiii) [reply src/py/pyodide/http.py:211]: It looks like just `**kwargs` is fine and doesn't mess with rst, but could be wrong. https://stackoverflow.com/questions/62511086/how-to-document-kwargs-according-to-numpy-style-docstring

Comment (hoodmane) [reply src/py/pyodide/http.py:211]: I suspect that I added the backslashes to fix some issue with the generated output but I have repeatedly been wrong about this sort of thing so there's a good chance that they are pointless.

**Pull request closed without merging**